### PR TITLE
review and update the android plugin migration doc

### PR DESCRIPTION
**Pull request by xster**
Fixes https://github.com/flutter/flutter/issues/44224

### CODE REVIEW

File: src/docs/development/packages-and-plugins/plugin-api-migration.md [hunk 1]
```diff
@@ -3,25 +3,25 @@ title: Supporting the new Android plugins APIs
...
+deprecated, but we encourage you to migrate to the new APIs based on
+[FlutterPlugin][].
 
-As of the 1.10.17 release, new plugin APIs are available on the
-master and dev channels. The old APIs won't be immediately
-deprecated, but we encourage you to migrate to the new APIs.
-Over time, plugins using the old APIs might produce strange
-behaviors when embedding Flutter into an Android app.
+The new APIs has the advantage of providing a cleaner set of accessors for
+lifecycle dependent components compared to the old APIs. For instance
```
Comment (gaaclarke) [inline src/docs/development/packages-and-plugins/plugin-api-migration.md]: s/APIs/API

File: src/docs/development/packages-and-plugins/plugin-api-migration.md [hunk 2]
```diff
@@ -32,70 +32,51 @@ a plugin that uses the new APIs, see the
...
+    public class MainActivity extends FlutterActivity {
+      // You can keep this empty class or remove it. Plugins on the new embedding
+      // now automatically registers plugins.
+    }
+    ```
 
-6. Create an `EmbeddingV1Activity.java` file that uses the v1
-   embedding in the same folder as `MainActivity`. For example:
+1. (Optional) Create an `EmbeddingV1Activity.java` file that uses the v1
+   embedding in the same folder as `MainActivity` in the example project to keep
```
Comment (gaaclarke) [inline src/docs/development/packages-and-plugins/plugin-api-migration.md]: Awkward.  "Create an `EmbeddingV1Activity.java` file that uses the v1 embedding for the example project in the same folder as `MainActivity` to keep..."

File: src/docs/development/packages-and-plugins/plugin-api-migration.md [hunk 3]
```diff
@@ -113,7 +94,7 @@ public class EmbeddingV1Activity extends FlutterActivity {
 }
 ```
 
-7. Add the `EmbeddingV1Activity` to the
+7. (Optional) Add the `EmbeddingV1Activity` to the
```
Comment (gaaclarke) [inline src/docs/development/packages-and-plugins/plugin-api-migration.md]: For these optional steps should we mention when they would want to be done?

File: src/docs/development/packages-and-plugins/plugin-api-migration.md [hunk 4]
```diff
@@ -3,25 +3,25 @@ title: Supporting the new Android plugins APIs
...
+The old APIs based on [PluginRegistry.Registrar][] won't be immediately
+deprecated, but we encourage you to migrate to the new APIs based on
+[FlutterPlugin][].
 
-As of the 1.10.17 release, new plugin APIs are available on the
-master and dev channels. The old APIs won't be immediately
-deprecated, but we encourage you to migrate to the new APIs.
-Over time, plugins using the old APIs might produce strange
-behaviors when embedding Flutter into an Android app.
+The new APIs has the advantage of providing a cleaner set of accessors for
```
[the remark this answers is not in the document]
Comment (xster) [reply src/docs/development/packages-and-plugins/plugin-api-migration.md]: ok

File: src/docs/development/packages-and-plugins/plugin-api-migration.md (hunk 1 above)
Comment (xster) [reply src/docs/development/packages-and-plugins/plugin-api-migration.md]: ok

File: src/docs/development/packages-and-plugins/plugin-api-migration.md (hunk 2 above)
Comment (xster) [reply src/docs/development/packages-and-plugins/plugin-api-migration.md]: ok

File: src/docs/development/packages-and-plugins/plugin-api-migration.md (hunk 3 above)
Comment (xster) [reply src/docs/development/packages-and-plugins/plugin-api-migration.md]: added a few more words

**Comment by sfshaza2** [comment]
@xster, it looks like you forked this when the build was broken. Would you mind grabbing the latest, merging, and submitting again? The build is fixed now. Thz

**Comment by xster** [comment]
CI passes now

**Comment by sfshaza2** [comment]
Waiting for @mklim, @gaaclarke, and/or @matthew-carroll to review.

File: src/docs/development/packages-and-plugins/plugin-api-migration.md [hunk 5]
```diff
@@ -169,138 +117,137 @@ but aren't required.
...
    For example:
 
-```yaml
-environment:
-  sdk: ">=2.0.0-dev.28.0 <3.0.0"
-  flutter: ">=1.9.1+hotfix.4 <2.0.0"
+    ```yaml
+    environment:
+      sdk: ">=2.0.0-dev.28.0 <3.0.0"
+      flutter: ">=1.9.1+hotfix.4 <2.0.0"
```
Comment (mklim) [inline src/docs/development/packages-and-plugins/plugin-api-migration.md]: I think this should be the new stable now, 1.12. 1.9 is missing some fixes that will matter in a few edge cases so I think it's better to just recommend stable here.

File: src/docs/development/packages-and-plugins/plugin-api-migration.md [hunk 6]
```diff
@@ -169,138 +117,137 @@ but aren't required.
...
-  sdk: flutter
-```
+    ```yaml
+    e2e: ^0.2.1
+    flutter_driver:
+      sdk: flutter
+    ```
 
-4. Manually register the E2E plugin in `MainActivity.java`
+1. Manually register the E2E plugin in `MainActivity.java`
```
Comment (mklim) [inline src/docs/development/packages-and-plugins/plugin-api-migration.md]: Is this still needed in v1.12?

**Pull request merged**